catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -195,6 +198,27 @@ public static String generateRemixUrlsStringForMergedProgram(XmlHeader headerOfF
 		return extractedUrls;
 	}
 
+	public static String getFileNameFromURL(String url) {
+		int extension = url.lastIndexOf('.');
+		int path = Math.max(url.substring(0, extension).lastIndexOf('/'), 0) + 1;
+		int param1 = url.indexOf('&', extension);
+		int param2 = url.indexOf('?', extension);
+		int param = (param2 >= 0 && param2 < param1) || (param1 == -1) ? param2 : param1;
+		return param >= 0 ? url.substring(path, param) : url.substring(path);
```

Comment:
Could you please think of better namings here. This is not clean code like and hard to follow what its doing.